src/BinaryFile.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Php\Pie;
+
+use function file_get_contents;
+use function hash;
+
+/**
+ * @internal This is not public API for PIE, so should not be depended upon unless you accept the risk of BC breaks
+ *
+ * @immutable
+ */
+final class BinaryFile
+{
+    private const HASH_TYPE_SHA256 = 'sha256';
+
+    /**
+     * @param non-empty-string $filePath
+     * @param non-empty-string $checksum
+     */
+    public function __construct(
+        public readonly string $filePath,
+        public readonly string $checksum,
+    ) {
+    }
+
+    /** @param non-empty-string $filePath */
+    public static function fromFileWithSha256Checksum(string $filePath): self
+    {
+        return new self(
+            $filePath,
+            hash(self::HASH_TYPE_SHA256, file_get_contents($filePath)),
```

Comment:
Could be `hash_file()` to not load the entire file contents into memory.
```suggestion
            hash_file(self::HASH_TYPE_SHA256, $filePath),
```

Reply:
Thanks!